src/Cns.ts

```diff
@@ -346,6 +342,38 @@ export default class Cns extends NamingService {
     const lastResetEvent = logs[logs.length - 1];
     return lastResetEvent?.blockNumber || defaultStartingBlock;
   }
+
+  private checkNetworkConfig(source: CnsSource): void {
+    if (!source.network) {
+      throw new ConfigurationError(ConfigurationErrorCode.UnspecifiedNetwork, {
+        method: NamingServiceName.CNS,
+      });
+    }
+    if (!CnsSupportedNetwork.guard(source.network)) {
+      this.checkCustomNetworkConfig(source);
+    }
+  }
+
+  private checkCustomNetworkConfig(source: CnsSource): void {
+    if (!source.proxyReaderAddress) {
+      throw new ConfigurationError(
+        ConfigurationErrorCode.CustomNetworkConfigMissing,
+        {
+          method: NamingServiceName.CNS,
+          config: 'proxyReaderAddress',
+        },
+      );
+    }
+    if (!source['url'] && source['provider']) {
```

Comment:
Should it look like this?

```suggestion
    if (!source['url'] || !source['provider']) {
```

Reply:
I think it should look like !source['url'] && !source['provider'] so that either can be defined. Provider is what's required for network calls and can be created from a url.